services/search.py

```
@@ -0,0 +1,570 @@
+def rank_results(results: list[SearchResult]) -> list[SearchResult]:
+    """Sort *results* by timestamp descending.
+
+    Handles both integer epoch-ms timestamps and ISO 8601 strings so the
+    three source types (composer, chat, cli_agent) sort together correctly.
+    """
+    def _ts(r: SearchResult) -> float:
+        t = r.get("timestamp", 0)
+        if isinstance(t, str):
+            try:
+                return datetime.fromisoformat(t.replace("Z", "+00:00")).timestamp()
+            except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)